arangod/GeneralServer/H2CommTask.cpp

```diff
@@ -580,9 +581,24 @@ void H2CommTask<T>::sendResponse(std::unique_ptr<GeneralResponse> res,
 
   // this uses a fixed capacity queue, push might fail (unlikely, we limit max streams)
   unsigned retries = 512;
-  while (ADB_UNLIKELY(!_responses.push(tmp))) {
-    std::this_thread::yield();
-    if (--retries == 0) {
+  try {
```

Comment:
We seem to have the same handling for when an exception is thrown and when `retries` reaches zero.
We could thus change the code to:
```
unsigned retries = 512;
try {
  while (ADB_UNLIKELY(!_responses.push(tmp)) && --retries > 0) {
    std::this_thread::yield();
  }
} catch (...) {
  retries = 0;
}
if (retries == 0) {
  LOG_TOPIC("924dc", WARN, Logger::REQUESTS) << "was not able to queue response" << (void*)this;
  // we are overloaded close stream
  asio_ns::post(this->_protocol->context.io_context,
                [self(this->shared_from_this()), mid(res->messageId())] {
                  auto& me = static_cast<H2CommTask<T>&>(*self);
                  nghttp2_submit_rst_stream(me._session, NGHTTP2_FLAG_NONE,
                                            static_cast<int32_t>(mid),
                                            NGHTTP2_ENHANCE_YOUR_CALM);
                });
  return;
}